fathom/src/lang/core.rs

```diff
@@ -83,6 +83,8 @@ pub struct StructType {
     pub doc: Arc<[String]>,
     /// Name of this definition.
     pub name: String,
+    /// Parameter telescope.
+    pub params: Arc<[(Located<String>, Arc<Term>)]>,
```

Comment:
Ahh, I guess this isn't actually a telescope just yet - the parameters can't depend on each other yet! We'd need to implement dependent function types first (see #281).